core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanOps.scala

```diff
@@ -74,7 +106,8 @@ trait SpanOps[F[_]] {
     *   }
     *   }}}
     */
-  def use[A](f: Result => F[A]): F[A]
+  final def use[A](f: Span[F] => F[A])(implicit F: MonadCancelThrow[F]): F[A] =
```

Comment:
This change will break quite a lot of code, and I would prefer to preserve the original behavior. 

In the previous version, effect constraints were hidden in the implementation, and a user could use `Tracer[F]` without additional constraints:
```scala
class Service[F[_]: Tracer] {
  def dbCall() = 
    Tracer[F].span("db-call").use(span => ...)
}
```

Now, a new constraint is mandatory:
```scala
class Service[F[_]: Tracer: MonadCancelThrow] {
  def dbCall() = 
    Tracer[F].span("db-call").use(span => ...)
}
```

There are better examples than this, but a few libraries that were instrumented before would struggle with this change.

Reply:
fair enough. I'll remove this change